src/OmniSharp.Host/OmniSharpCommandLineApplication.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Internal;
+
+namespace OmniSharp
+{
+    public class OmniSharpCommandLineApplication
+    {
+        protected readonly CommandLineApplication Application;
+        private readonly CommandOption _hostPid;
+        private readonly CommandOption _stdio;
+        private readonly CommandOption _zeroBasedIndices;
+        private readonly CommandOption _encoding;
+        private readonly CommandOption _plugin;
+        private readonly CommandOption _verbose;
+        private readonly CommandOption _logLevel;
+        private readonly CommandOption _applicationRoot;
+
+        public OmniSharpCommandLineApplication()
+        {
+            Application = new CommandLineApplication(throwOnUnexpectedArg: false);
+            Application.HelpOption("-? | -h | --help");
+
+            _applicationRoot = Application.Option("-s | --source", "Solution or directory for OmniSharp to point at (defaults to current directory).", CommandOptionType.SingleValue);
+            _logLevel = Application.Option("-l | --loglevel", "Level of logging (defaults to 'Information').", CommandOptionType.SingleValue);
+            _verbose = Application.Option("-v | --verbose", "Explicitly set 'Debug' log level.", CommandOptionType.NoValue);
+            _hostPid = Application.Option("-hpid | --hostPID", "Host process ID.", CommandOptionType.SingleValue);
+            _stdio = Application.Option("-stdio | --stdio", "Use STDIO over HTTP as OmniSharp commincation protocol.", CommandOptionType.NoValue);
+            _stdio = Application.Option("-lsp | --lsp", "Use Language Server Protocol.", CommandOptionType.NoValue);
```

Comment:
Oh, I guess setting `-lsp` implicitly sets `_stdio`? Might be worth a source comment.